### PR TITLE
planner: simplify IsAutoCommitNotValidTxn

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
`IsAutoCommitNotValidTxn` is a bit complex.

### What is changed and how it works?
Use `InTxn()` instead of `Valid()` to check the transaction is auto-commit or not.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

File: planner/core/common_plans.go [hunk 1]
```diff
@@ -904,11 +904,7 @@ func (e *Explain) prepareTaskDot(p PhysicalPlan, taskTp string, buffer *bytes.Bu
 //  2. txn is not valid
 //  3. plan is point get by pk, or point get by unique index (no double read)
 func IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx sessionctx.Context, p Plan) (bool, error) {
-	ok, err := IsAutoCommitNonValidTxn(ctx)
-	if err != nil {
-		return false, err
-	}
-	if !ok {
+	if !isAutoCommitTxn(ctx) {
```
Comment (cfzjywxk) [inline planner/core/common_plans.go]: Better to update the function comments above

Comment (jackysp) [reply planner/core/common_plans.go]: Done.

**Comment by cfzjywxk** [comment]
LGTM

**Pull request merged**